### ISSUE TITLE
Remove the newline at the end of each file

### ISSUE THREAD

**Issue by kalbasit** [darold/pgFormatter#196]
Hello,

When I run the pgFormat, it seems to add a new line at the end of each file. This is conflicting with a pre-commit hook that we have that removes the newline from the end of the file. Is it possible to stop adding that?

**Comment by darold** [comment]
Hi,

If I have well understood what you've reported commit f71247f might fix this issue.

**Comment by polobo** [comment]
What is the goal here?  To preserve everything after the final semi-colon as-is or ensure a standard output format?

Looking at the test file ex0.sql the input file contains two empty lines at the end.  The expected output went from having one empty line to zero empty lines.  If the goal is preservation I would have expected two empty lines in the expected output file, not zero or one.  If you simply want to ensure consistent output then the choice between an empty line at the end or not is really a choice that the user should be allowed to make.  Since there is need for a default choice the choice of a trailing empty line (i.e., newline) is no less valid than not having a trailing newline so changing the default from one to the other seems unwarranted.

In short I don't think this commit is a good solution - documenting the goal and then having a few tests with various end of file line combinations demonstrating the final result would be better.  Absent that work I'd say the original behavior should be preserved and it wasn't wrong and this is arguably strictly worse.

Then, if going for preservation the expected outputs for most of these files will change given they almost all seem to have no trailing newline.  But the expected output for ex0.sql should add two newlines instead of removing one.

**Comment by darold** [comment]
Actually pgFormatter is already modifying the original file content as it always adds a single empty line after a `;`.   So even before this commit the two empty lines at end of ex0.sql was changed, pgFormatter has never preserved the exact content of the source file.

If this is for myself I prefer to have an extra empty line at end of the output but I know that this is not common and the extra line can be useless. @polobo is there any reason why you want to keep this extra line? I see two good reasons, the first one is backward compatibility and the second one is having a separator line with other content following, for example when using pgFormatter in an editor.

I can keep thing like that and add an option to remove the extra empty line if there is a consensus.

**Comment by polobo** [comment]
Backward compatibility seems sufficient enough a reason.  Adding a option so the user can choose is the ideal way to accommodate the request.

**Comment by darold** [comment]
After more though I totaly agree, I have reverted the commit and working on adding  `--no-extra-line` command line option. Thanks for the review, much appreciated.

**Comment by darold** [comment]
Then available in commit 91983eb

**Comment by kalbasit** [comment]
@darold thank you for the quick turn around!! 91983ebfa9e43f37eebfaa6adb30f09568d35ec4 works for me!
